Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -0,0 +1,645 @@
+        }
+
+        composerView.inputMessageView.content = .init(
+            quotingMessage: content.quotingMessage,
+            command: content.command,
+            documentAttachments: content.documentAttachments,
+            imageAttachments: content.imageAttachments
+        )
+
+        if content.isInsideThread {
+            if channelController.channel?.isDirectMessageChannel == true {
+                composerView.checkboxControl.label.text = L10n.Composer.Checkmark.directMessageReply
+            } else {
+                composerView.checkboxControl.label.text = L10n.Composer.Checkmark.channelReply
+            }
+        }
+        Animate {
+            self.composerView.bottomContainer.isHidden = !self.content.isInsideThread
+        }
+
+        if let typingCommand = typingCommand(in: composerView.inputMessageView.textView) {
+            showCommandSuggestions(for: typingCommand)
+            return
+        }
+
+        if let (typingMention, mentionRange) = typingMention(in: composerView.inputMessageView.textView) {
+            showMentionSuggestions(for: typingMention, mentionRange: mentionRange)
+            return
+        }
+
+        dismissSuggestions()
+    }
+    
+    // MARK: - Actions
+    
+    @objc open func publishMessage(sender: UIButton) {
+        let text: String
+        if let command = content.command {
+            text = "/\(command.name) " + content.text
+        } else {
+            text = content.text
+        }
+
+        if let editingMessage = content.editingMessage {
+            editMessage(withId: editingMessage.id, newText: text)
+        } else {
+            createNewMessage(text: text)
+        }
+
+        content = .empty()
+    }
+    
+    @objc open func showAttachmentsPicker(sender: UIButton) {
+        let actionSheet: UIAlertController = UIAlertController(
+            title: nil,
+            message: L10n.Composer.Picker.title,
+            preferredStyle: .actionSheet
+        )
+        actionSheet.popoverPresentationController?.sourceView = sender
+
+        let showFilePickerAction = UIAlertAction(
+            title: L10n.Composer.Picker.file,
+            style: .default,
+            handler: { [weak self] _ in
+                guard let self = self else { return }
+                self.present(self.filePickerVC, animated: true, completion: nil)
+            }
+        )
+        let showImagePickerAction = UIAlertAction(
+            title: L10n.Composer.Picker.image,
+            style: .default,
+            handler: { [weak self] _ in
+                guard let self = self else { return }
+                self.present(self.imagePickerVC, animated: true, completion: nil)
+            }
+        )
+        let cancelAction = UIAlertAction(title: L10n.Composer.Picker.cancel, style: .cancel)
+
+        [showFilePickerAction, showImagePickerAction, cancelAction].forEach {
+            actionSheet.addAction($0)
+        }
```

Comment:
This was a computed variable before, but now it's initialised every time the user tap to add an attachment, even when the `actionSheet` is not needed anymore, please change it back to save some time, opening of the image or file picker is slow enough 😅

Reply:
Why would it save time if it was a computed variable? 😄 That is just syntax sugar. A computed variable is not a lazy variable, it will always return a new `UIAlertController`. But what you are saying maybe makes sense, we probably can reuse the `UIAlertController`, but for that, we should have a lazy ref just like we have with the other view controllers we have in this component.

Reply:
A computed variable is computed at the time it's accessed, and not computed if it's not accessed.